File: app/src/main/java/com/example/schedule/ui/AboutLessonActivity.java
package com.example.schedule.ui;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.schedule.R;
import com.example.schedule.model.Lesson;

import static com.example.schedule.ScheduleApplication.COLOR_ACCENT;
import static com.example.schedule.ScheduleApplication.COLOR_PRIMARY;
import static com.example.schedule.ScheduleApplication.COLOR_SECONDARY;
import static com.example.schedule.ScheduleApplication.currentTheme;

public class AboutLessonActivity extends AppCompatActivity {

    Lesson lesson;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        int theme = R.style.AppTheme_NoActionBar;
        switch(currentTheme) {
            case R.style.AppTheme_Dark:
                theme = R.style.AppTheme_Dark_NoActionBar;
                break;
            case R.style.AppTheme_Light:
                theme = R.style.AppTheme_Light_NoActionBar;
                break;
        }

        setTheme(theme);
        setContentView(R.layout.activity_about_lesson);

        setTitle("О предмете");

        // Set toolbar
        Toolbar toolbar = findViewById(R.id.toolbar);
        toolbar.setBackgroundColor(getResources().getColor(COLOR_PRIMARY));
        toolbar.setTitleTextColor(getResources().getColor(android.R.color.white));
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);
        toolbar.setNavigationOnClickListener(view -> onBackPressed());

        RecyclerView recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));
        recyclerView.setAdapter(new AboutLessonAdapter());

        lesson = getIntent().getParcelableExtra("lesson");
    }

    class AboutLessonAdapter extends RecyclerView.Adapter<AboutLessonAdapter.LessonViewHolder> {

        @NonNull
        @Override
        public LessonViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_about_lesson, parent, false);

            return new LessonViewHolder(view);
        }

        @Override
        public void onBindViewHolder(@NonNull LessonViewHolder holder, int position) {
            holder.bind(position);
        }

        @Override
        public int getItemCount() {
            return Lesson.fieldsCount;
        }

        class LessonViewHolder extends RecyclerView.ViewHolder {

            TextView twName, twValue;

            LessonViewHolder(@NonNull View itemView) {
                super(itemView);

                twName = itemView.findViewById(R.id.twName);
                twValue = itemView.findViewById(R.id.twValue);

                twName.setTextColor(itemView.getContext().getResources().getColor(COLOR_SECONDARY));
                twValue.setTextColor(itemView.getContext().getResources().getColor(COLOR_ACCENT));
            }

            void bind(int position) {
                twName.setText(lesson.getNames().get(position));
                twValue.setText(lesson.getValues().get(position));
            }
        }
    }
}

File: app/src/main/java/com/example/schedule/ui/AboutLessonActivity.kt
package com.example.schedule.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.schedule.R
import com.example.schedule.ScheduleApplication
import com.example.schedule.model.Lesson
import com.example.schedule.util.MySimpleAdapter
import kotlinx.android.synthetic.main.activity_about_lesson.*
import kotlinx.android.synthetic.main.item_about_lesson.view.*

class AboutLessonActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        var theme = R.style.AppTheme_NoActionBar
        when (ScheduleApplication.currentTheme) {
            R.style.AppTheme_Dark -> theme = R.style.AppTheme_Dark_NoActionBar
            R.style.AppTheme_Light -> theme = R.style.AppTheme_Light_NoActionBar
        }
        setTheme(theme)
        setContentView(R.layout.activity_about_lesson)
        title = "О предмете"

        // Set toolbar
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        toolbar.setBackgroundColor(resources.getColor(ScheduleApplication.COLOR_PRIMARY))
        toolbar.setTitleTextColor(resources.getColor(android.R.color.white))
        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setDisplayShowHomeEnabled(true)
        toolbar.setNavigationOnClickListener { onBackPressed() }

        val lesson = intent.extras!!.get(ScheduleApplication.LESSON_EXTRA) as Lesson
        rvLessonInfo.layoutManager = LinearLayoutManager(this)
        rvLessonInfo.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))
        rvLessonInfo.adapter = MySimpleAdapter(R.layout.item_about_lesson, lesson.fields, { itemView, lessonField ->
            with(itemView) {
                twName.text = lessonField.name
                twValue.text = lessonField.value
            }
        }, {
            with (it) {
                twName.setTextColor(context.resources.getColor(ScheduleApplication.COLOR_SECONDARY))
                twValue.setTextColor(context.resources.getColor(ScheduleApplication.COLOR_ACCENT))
            }
        })
    }
}